app/javascript/pages/country/widget/widgets/widget-tree-gain/widget-tree-gain.js

```diff
@@ -8,13 +8,35 @@ import { getActiveFilter } from 'pages/country/widget/widget-selectors';
 
 import actions from './widget-tree-gain-actions';
 import reducers, { initialState } from './widget-tree-gain-reducers';
+import { getSortedData } from './widget-tree-gain-selectors';
 import WidgetTreeGainComponent from './widget-tree-gain-component';
 
-const mapStateToProps = ({ widgetTreeGain }, ownProps) => ({
-  loading: widgetTreeGain.loading || ownProps.isMetaLoading,
-  gain: widgetTreeGain.data.gain,
-  extent: widgetTreeGain.data.extent
-});
+const mapStateToProps = ({ location, widgetTreeGain, countryData }) => {
+  const { isCountriesLoading, isRegionsLoading } = countryData;
+  let meta = [];
+  if (location.payload.subRegion) {
+    meta = countryData.subRegions;
+  } else if (location.payload.region) {
+    meta = countryData.regions;
+  } else {
+    meta = countryData.countries;
+  }
+
+  return {
+    loading: widgetTreeGain.loading || isCountriesLoading || isRegionsLoading,
+    data: {
+      gain: widgetTreeGain.data.gain,
+      extent: widgetTreeGain.data.extent,
```

Comment:
I have been playing with putting the dynamic sentence calculation inside the selectors. Seeing as we a re creating complicated structures for getting the data in the correct format, maybe it is easier this. Then we don't have to keep passing the data around, and we can cache the data to reduce recalculation. Thoughts?